packages/server/modules/workspaces/tests/integration/projects.graph.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { AllScopes } from '@/modules/core/helpers/mainConstants'
+import {
+  BasicTestUser,
+  createAuthTokenForUser,
+  createTestUser
+} from '@/test/authHelper'
+import {
+  CreateWorkspaceDocument,
+  CreateWorkspaceProjectDocument,
+  GetWorkspaceProjectsDocument,
+  Workspace
+} from '@/test/graphql/generated/graphql'
+import {
+  createTestContext,
+  testApolloServer,
+  TestApolloServer
+} from '@/test/graphqlHelper'
+import { beforeEachContext } from '@/test/hooks'
+import { Roles } from '@speckle/shared'
+import { expect } from 'chai'
+import cryptoRandomString from 'crypto-random-string'
+
+describe('Workspace project GQL CRUD', () => {
+  let apollo: TestApolloServer
+  let workspace: Omit<Workspace, 'projects' | 'team'>
+
+  const testUser: BasicTestUser = {
+    id: '',
+    name: 'John Speckle',
+    email: 'john-speckle-workspace-project-admin@example.org',
+    role: Roles.Server.Admin
+  }
+
+  const testNonWorkspaceMemberUser: BasicTestUser = {
+    id: '',
+    name: 'John Nobody',
+    email: 'john-nobody@example.org',
+    role: Roles.Server.User
+  }
+
+  before(async () => {
+    await beforeEachContext()
+    await Promise.all(
+      [testUser, testNonWorkspaceMemberUser].map((user) => createTestUser(user))
+    )
+    const token = await createAuthTokenForUser(testUser.id, AllScopes)
+    apollo = await testApolloServer({
+      context: createTestContext({
+        auth: true,
+        userId: testUser.id,
+        token,
+        role: testUser.role,
+        scopes: AllScopes
+      })
+    })
+
+    const { data } = await apollo.execute(CreateWorkspaceDocument, {
```

Comment:
Why not use the `createTestWorkspace` helper?

Reply:
I am silly and like to duplicate work (done)